flink-core/src/main/java/org/apache/flink/api/common/typeutils/CompositeTypeSerializerConfigSnapshot.java

```diff
@@ -73,13 +91,11 @@ public boolean equals(Object obj) {
 		}
 
 		return (obj.getClass().equals(getClass()))
-				&& Arrays.equals(
-					nestedSerializerConfigSnapshots,
-					((CompositeTypeSerializerConfigSnapshot) obj).getNestedSerializerConfigSnapshots());
+				&& nestedSerializersWithConfigs.equals(((CompositeTypeSerializerConfigSnapshot) obj).getNestedSerializersAndConfigs());
```

Comment:
I've changed `nestedSerializersWithConfigs` to a list, it's no longer an array. The `equals` implementation should suffice?